### ISSUE TITLE
Reactive Table doesn't update when key value changed (newbie)

### ISSUE THREAD

**Issue by RGologorsky** [aslagle/reactive-table#330]
When I update an attribute in a collection that I am using as a key, the change isn't registered in the reactive table, e.g. if the key is called 'date' and I update the value of date, the new value isn't updated in the reactive table although console logging the value of date shows that it has updated. Is there a fix? Thanks!

**Comment by aslagle** [comment]
That's supposed to work, I'm not sure what the problem could be. Do you have the same problem with other keys? Other collections?

**Comment by RGologorsky** [comment]
No, the problem is only with that particular key. I initially set the key value to -1, and then when the user selects a value for a month filter (e.g. Dec through Feb), I update the key value for all the matching results, and then set the filter. I think what happens is that the filter gets set while the values are still updating, and then doesn't re-update once the values are done updating. (For context, I do this so that if the filter is Dec to Feb the results are ordered Dec Jan Feb not Jan Feb Dec.)

**Comment by aslagle** [comment]
I don't understand why you're updating the collection. I wouldn't add a column if you already have the dates somewhere else in the collection, or else I'd always set the values to the dates. I'd use $gt and $lt filters to limit the table to the right months, and then sort by year before month to put Dec before Jan.

**Comment by RGologorsky** [comment]
I wasn't able to get years from the data, which is why I resorted to this work-around. I realize in retrospect it probably wasn't the best way. Thank you though for your idea on filtering dates by month value using $gt and $lt.  

Anyways, the problem does remain though that the collection doesn't always update when I update the date -- usually the problem is when I set the filter for a long time span like March - August. If you would like to see the code behind the problem, I posted my code on GitHub, https://github.com/RGologorsky/testing; the faulty key is dateRank, which is a compilation of month & date that I use to sort by and the relevant files are monthFilter.js and tournsTable.js in the client folder. (Forgive the sloppy code, I'm a newbie who did this as a side project for fun over a few days during winter break.) I deployed the site to http://searchchesstournaments.meteor.com/ if you want to test it out; seeing "Loading" in the column means that the reactive table hasn't yet registered that dateRank has not been updated from -1 to the actual date. 

Thank you for looking into this.

**Comment by aslagle** [comment]
Thanks for sharing the code. I haven't figured out the problem, but I think it's related to using collection.find as the collection. For some reason, the way I'm observing changes isn't working in this case. If you make zip a filter and take out the find, I think it will fix it.

**Comment by RGologorsky** [comment]
It worked, thank you!
